src/source/tcp.rs

```diff
@@ -151,23 +194,28 @@ where
         loop {
             match listener.accept() {
                 Ok((stream, _addr)) => {
+                    // Actually spawn the stream handler
                     let rchans = chans.to_owned();
-                    let new_stream = thread::spawn(move |poller| {
-                        // Note - Stream handlers are allowed to crash without
-                        // compromising Cernan's ability to gracefully shutdown.
-                        poller
-                            .register(
-                                &stream,
-                                mio::Token(0),
-                                mio::Ready::readable(),
-                                mio::PollOpt::edge(),
-                            )
-                            .unwrap();
-
-                        let mut handler = H::new();
-                        handler.handle_stream(rchans, &poller, stream);
-                    });
-                    self.handlers.push(new_stream);
+                    let stream_events_readiness = self.stream_events_readiness.clone();
```

Comment:
Something seems off with this. We're cloning this thing here and passing it in, but we're cloning it again in spawn2... how is this meant to be used?